api/src/utils/githubhandler.js

```diff
@@ -0,0 +1,75 @@
+import Ajv from 'ajv';
+import schema from '../schemas/topic.json';
+import { createNewRefFromBase, createFile, createPullRequest, updateFile } from './github';
+
+const ajv = new Ajv();
+
+const validate = ajv.compile(schema);
+
+/**
+ *
+ * @param {String} operation
+ * @param {String} defaultBranch
+ * @param {String} repo
+ * @param {String} owner
+ * @param {String} req
+ * @param {String} ref
+ * @param {String} bodyData
+ * @param {String} topicName
+ * @returns {Object} {response {status, message, prUrl}}
+ */
+export const createPullRequestFromData = async (
+  operation,
+  defaultBranch,
+  repo,
+  owner,
+  req,
+  ref,
+  bodyData,
+  topicName,
+) => {
+  let response = { statusMessage: 'Ok', prUrl: '', status: '200' };
+  try {
+    // create new topic and branch
+    // validate topic schema
+    const isValidData = validate(req.body);
+    if (isValidData) {
+      // TODO -- validate topic doesn't already exist
+      // TODO -- validate topic sources are valid
+      // create a git branch on the remote with a naming convention [createTopic/topicname]
+      await createNewRefFromBase(owner, repo, ref);
+      // // create a new file with contents
+      if (operation === 'create') {
+        await createFile(owner, repo, bodyData, ref, topicName);
+      } else if (operation === 'update') {
+        await updateFile(owner, repo, bodyData, ref, topicName);
+      }
+      // commit  to branch
+      // make pr against ref to base using templates
+      const topicDescription = JSON.parse(bodyData).description;
+      const pullRequest = await createPullRequest(
+        operation,
+        owner,
+        repo,
+        defaultBranch,
+        topicName,
+        ref,
+        topicDescription,
+      );
+      // URL to the pull Request created ..
+      const pullRequestUrl = pullRequest.data.html_url;
```

Comment:
this next line could be collapsed into one. This just makes things less verbose / more readable